### PR TITLE
DRA: generate deepcopy for DriverResources

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This PR introduces deepcopy-gen fiels for `staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.{DriverResources,Pool}`, introducing the deepcopy interfaces for said types at `staging/src/k8s.io/dynamic-resource-allocation/resourceslice/zz_generated.deepcopy.go`

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by klueska** [comment]
/hold

Can you also put the deepcopy gen on the `Owner` type for completeness.

**Comment by ArangoGutierrez** [comment]
> /hold
> 
> Can you also put the deepcopy gen on the `Owner` type for completeness.

Done

**Review by hashim21223445 — APPROVED** [review]

**Review by hashim21223445 — APPROVED** [review]

File: staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go [hunk 1]
```diff
@@ -73,12 +73,14 @@ type Controller struct {
 	resources *DriverResources
 }
 
+// +k8s:deepcopy-gen=true
```
Comment (pohly) [inline staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go:76]: /lgtm cancel

Please insert an empty line between this special comment and the normal Go docs.

File: staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go [hunk 2]
```diff
@@ -94,6 +96,7 @@ type Pool struct {
 	Devices []resourceapi.Device
 }
 
+// +k8s:deepcopy-gen=true
```
Comment (pohly) [inline staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go:99]: Same here,

File: staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go [hunk 3]
```diff
@@ -73,12 +73,14 @@ type Controller struct {
...
 // DriverResources is a complete description of all resources synchronized by the controller.
 type DriverResources struct {
 	// Each driver may manage different resource pools.
 	Pools map[string]Pool
 }
 
+// +k8s:deepcopy-gen=true
```
Comment (pohly) [inline staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go:83]: And here.

File: staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go (hunk 1 above)
Comment (ArangoGutierrez) [reply staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go:76]: done

File: staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go (hunk 2 above)
Comment (ArangoGutierrez) [reply staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go:99]: done

File: staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go (hunk 3 above)
Comment (ArangoGutierrez) [reply staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go:83]: done

**Review by pohly — COMMENTED** [review]
/lgtm

**Pull request merged**